Assignment.Tests/PingProcess.Tests.cs

```diff
@@ -0,0 +1,232 @@
+    public void TestInitialize()
+    {
+        Sut = new();
+    }
+
+    [TestMethod]
+    public void Start_PingProcess_Success()
+    {
+        Process process = Process.Start("ping", "localhost");
+        process.WaitForExit();
+        Assert.AreEqual<int>(0, process.ExitCode);
+    }
+
+    [TestMethod]
+    public void Run_GoogleDotCom_Success()
+    {
+        PingResult result = Sut.Run("google.com");
+        Assert.AreEqual<int>(0, result.ExitCode);
+    }
+
+
+    [TestMethod]
+    public void Run_InvalidAddressOutput_Success()
+    {
+        (int exitCode, string? stdOutput) = Sut.Run("badaddress");
+        Assert.IsFalse(string.IsNullOrWhiteSpace(stdOutput));
+        stdOutput = WildcardPattern.NormalizeLineEndings(stdOutput!.Trim());
+        Assert.AreEqual<string?>(
+            "Ping request could not find host badaddress. Please check the name and try again.".Trim(),
+            stdOutput,
+            $"Output is unexpected: {stdOutput}");
+        Assert.AreEqual<int>(1, exitCode);
+    }
+
+    [TestMethod]
+    public void Run_CaptureStdOutput_Success()
+    {
+        PingResult result = Sut.Run("localhost");
+        AssertValidPingOutput(result);
+    }
+
+    [TestMethod]
+    public void RunTaskAsync_Success()
+    {
+        PingProcess pingProcess = new();
+        string hostNameOrAddress = "localhost";
+
+
+        var pingTask = PingProcess.RunTaskAsync(hostNameOrAddress);
+
+        var result = pingTask.Result;
+        Assert.IsNotNull(result.StdOutput);
+        Assert.IsTrue(result.ExitCode == 0);
+    }
+    [TestMethod]
+    public void RunAsync_UsingTaskReturn_Success()
+    {
+
+        PingProcess pingProcess = new();
+        string hostNameOrAddress = "localhost";
+
+
+        var task = PingProcess.RunTaskAsync(hostNameOrAddress);
+        var res = task.Result;
+
+
+        Assert.IsNotNull(res.StdOutput);
+        Assert.IsTrue(res.ExitCode == 0);
+    }
+
+    [TestMethod]
+
+    async public Task RunAsync_UsingTpl_Success()
+    {
+
+        PingProcess pingProcess = new();
+        string hostNameOrAddress = "localhost";
+
+
+        var result = await PingProcess.RunAsync(hostNameOrAddress);
+
+
+        Assert.IsNotNull(result.StdOutput);
+        Assert.IsTrue(result.ExitCode == 0);
+    }
+
+
+
+    [TestMethod]
+    public void RunAsync_UsingTplWithCancellation_CatchAggregateExceptionWrapping()
+    {
+        PingProcess pingProcess = new();
+        string hostNameOrAddress = "localhost";
+        CancellationTokenSource cancellationTokenSource = new();
+
+        var task = PingProcess.RunAsync(hostNameOrAddress, cancellationTokenSource.Token);
+        cancellationTokenSource.Cancel();
+
+        try
+        {
+            task.Wait();
+            Assert.Fail("Expected an AggregateException to be thrown.");
+        }
+        catch (AggregateException ex)
+        {
+            Assert.IsTrue(ex.InnerExceptions.Any(innerEx => innerEx is TaskCanceledException));
```

Comment:
```suggestion

            Assert.ThrowsException<AggregateException>(() => task.Wait());

```
This method is checking if an `Aggregate Exception` is thrown. So having it also check the inner exception to make sure its the correct type isn't needed.